compiler/src/dotty/tools/dotc/core/JavaNullInterop.scala

```diff
@@ -260,38 +168,50 @@ object JavaNullInterop {
     }.toMap
   }
 
-  private def stdLibPolicy(sym: Symbol, tp: Type)(implicit ctx: Context): NullifyPolicy = {
+  private def isInNotNullStdLibList(sym: Symbol, tp: Type)(implicit ctx: Context): Boolean = {
     val ownerName = sym.owner.showFullName
-    if (!nullStats.contains(ownerName)) return FalsePolicy
+    if (!nullStats.contains(ownerName)) return false
+    
     val stats = nullStats(ownerName)
-    if (sym.is(Flags.Method)) {
+    if (sym.is(Flags.Method))
       stats.getMethod(sym, tp) match {
-        case Some(mstats) =>
-          def triggerAndLog(s: Symbol): Boolean = {
-//            if (mstats.nnRet) println(s">>> ${sym.name.show} with type ${tp.show}")
-            s == sym
-          }
-          MethodPolicy(triggerAndLog, /*mstats.nnParams*/ Seq.empty, mstats.nnRet)
-        case None => FalsePolicy
+        case Some(_) => true
+        case None => false
       }
-    } else {
+    else 
       stats.getField(sym, tp) match {
-        case Some(fstas) =>
-          NoOpPolicy(_ == sym)
-        case None => FalsePolicy
-      }
+        case Some(_) => true
```

Comment:
For fields in the list, we should still nullify the inside type without a `| JavaNull` at the outermost, similar to `@NotNulll` annotation on a field.

Reply:
Yes, you're right. The previous code was incorrect. Nice catch!